lib/std/mem.zig

```diff
@@ -1936,25 +1937,31 @@ pub fn nativeToBig(comptime T: type, x: T) T {
     };
 }
 
+fn CopyPtrAttrs(comptime source: type, size: builtin.TypeInfo.Pointer.Size, child: type) type {
+    const info = @typeInfo(source).Pointer;
+    return @Type(.{
+        .Pointer = .{
+            .size = size,
+            .is_const = info.is_const,
+            .is_volatile = info.is_volatile,
+            .is_allowzero = info.is_allowzero,
+            .alignment = info.alignment,
+            .child = child,
+            .sentinel = null,
+        },
+    });
+}
+
```

Comment:
I think something like the following is slightly better
```zig
var info = @typeInfo(source);
info.Pointer.child = child;
info.Pointer.size = size;
return @Type(info);
```

Reply:
I think my version makes it clearer what I am copying.